### PR TITLE
Fixed yggtorrents login issue

### PR DESCRIPTION
**Pull request by phtagn**
Fixes #

Proposed changes in this pull request:
- Fixes yggtorrent's login check logic
- Sanitize search string to remove characters that are not allowed on the site.
-

- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by PHD59fr** [comment]
I do not understand why you are trying to change the login verification system, we dont use 'find_all' against an 'in'.

When you call the page "https://yggtorrent.com/" , it will return the logged page with "logout" inside

```
root@NUNUX:~# python -m timeit 'temp = "https://yggtorrent.com/user/logout";temp.find("logout")'
10000000 loops, best of 3: 0.0929 usec per loop
root@NUNUX:~# python -m timeit 'temp = "https://yggtorrent.com/user/logout";"logout" in temp'
10000000 loops, best of 3: 0.0325 usec per loop
```
for the format of the search string, i think you can use urlencode

File: sickbeard/providers/yggtorrent.py [hunk 1]
```diff
@@ -48,10 +49,12 @@ def __init__(self):
         self.minleech = None
 
         # URLs
-        self.url = 'https://ww1.yggtorrent.com/'
+        self.url = 'https://yggtorrent.com/'
```
Comment (PHD59fr) [inline sickbeard/providers/yggtorrent.py:52]: OK for me

File: sickbeard/providers/yggtorrent.py [hunk 2]
```diff
@@ -48,10 +49,12 @@ def __init__(self):
...
         self.urls = {
             'login': urljoin(self.url, 'user/login'),
             'search': urljoin(self.url, 'engine/search'),
+            'login_check': urljoin(self.url, 'user/account'),
```
Comment (PHD59fr) [inline sickbeard/providers/yggtorrent.py]: Useless: login_check and logout

File: sickbeard/providers/yggtorrent.py [hunk 3]
```diff
@@ -71,12 +74,13 @@ def login(self):
 
         response = self.get_url(self.urls['login'], post_data=login_params, returns='text')
         if not response: # When you call /login if it's OK, it's return 200 with no body, i retry in main if it's logged !
-            response = self.get_url(self.url, returns='text')
+            response = self.get_url(self.urls['login_check'], returns='text')
```
Comment (PHD59fr) [inline sickbeard/providers/yggtorrent.py]: Test just on index, it's on for check:
`<li><a href="https://yggtorrent.com/user/logout"><i style="font-size:17px; vertical-align: middle;" class="fa fa-sign-out" aria-hidden="true"></i></a></li>`

File: sickbeard/providers/yggtorrent.py [hunk 4]
```diff
@@ -71,12 +74,13 @@ def login(self):
...
             if not response: # The provider is dead !!!
                 logger.log('Unable to connect to provider', logger.WARNING)
                 return False
 
-        if 'logout' not in response:
+        html = BS4Parser(response, 'html5lib')
```
Comment (PHD59fr) [inline sickbeard/providers/yggtorrent.py]: for perf use 'in', reponse is a string

File: sickbeard/providers/yggtorrent.py [hunk 5]
```diff
@@ -96,6 +100,18 @@ def search(self, search_strings, age=0, ep_obj=None):  # pylint: disable=too-man
                 if mode != 'RSS':
                     logger.log('Search string: {0}'.format
                                (search_string.decode('utf-8')), logger.DEBUG)
+                # search string needs to be normalized, single quotes are apparently not allowed on the site
```
Comment (PHD59fr) [inline sickbeard/providers/yggtorrent.py:100]: OK for me

**Review by PHD59fr — CHANGES_REQUESTED** [review]

**Comment by phtagn** [comment]
Hi,

I am not saying this is better. It stopped working and I thought it was because it didn’t find ‘logout’ on the page: I looked at couchpotato's yggtorrent searcher and this is the method they are using, so I thought I'd do the same. Didn't think to test for efficiency, sorry. 
I changed this and a few other things at the same time and I didn’t check what solved the issue, sorry. 
If you want I can revert to using ‘in’ that’s fine, or you can disregard the pull request entirely. Honestly, I now think it was just an URL issue, so changing from ww1.yggtorrent.com to yggtorrent.com may have solved it alone. 
What is necessary on the other hand is to sanitize the search string and remove apostrophe. Otherwise you can’t find any of the marvel shows as they’re named, for example, « Marvel’s Iron Fist ». Replacing ç with c also helps, when you’re searching for « fais pas ci, fais pas ça », but that does not happen very frequently.

**Comment by phtagn** [comment]
I’ve amended the file per your comments. Do you want to keep the sanitization part or should I delete it also ?

Kind regards

**Review by PHD59fr — APPROVED** [review]

File: sickbeard/providers/yggtorrent.py [hunk 6]
```diff
@@ -30,6 +30,7 @@
 from sickbeard.bs4_parser import BS4Parser
 from sickrage.helper.common import convert_size, try_int
 from sickrage.providers.torrent.TorrentProvider import TorrentProvider
+from lib.js2py.translators.pyjsparser import false
```
Comment (miigotu) [inline sickbeard/providers/yggtorrent.py]: ?

Comment (phtagn) [reply sickbeard/providers/yggtorrent.py]: Hi, sorry, did not realize that was in there. No reason for it to be, it was automatically generated by the IDE. I just removed it.

Comment (PHD59fr) [reply sickbeard/providers/yggtorrent.py]: It's OK now, can u merge the branch please ?

**Comment by miigotu** [comment]
Copied your commits to a new branch and fixed the build in #4239

**Pull request closed without merging**